storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCommon.java

```diff
@@ -0,0 +1,247 @@
+            return;
+        }
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_CLONE_CHILDREN), flag ? "1" : "0");
+    }
+
+    @Override
+    public boolean getCgroupCloneChildren() throws IOException {
+        return CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CGROUP_CLONE_CHILDREN)).get(0).equals("1") ? true : false;
+    }
+
+    @Override
+    public void setEventControl(String eventFd, String controlFd, String... args) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        sb.append(eventFd);
+        sb.append(' ');
+        sb.append(controlFd);
+        for (String arg : args) {
+            sb.append(' ');
+            sb.append(arg);
+        }
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_EVENT_CONTROL), sb.toString());
+    }
+
+    public Hierarchy getHierarchy() {
+        return hierarchy;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDir() {
+        return dir;
+    }
+
+    public CgroupCommon getParent() {
+        return parent;
+    }
+
+    public Set<CgroupCommon> getChildren() {
+
+        File file = new File(this.dir);
+        File[] files = file.listFiles();
+        if (files == null) {
+            return null;
+        }
+        Set<CgroupCommon> children = new HashSet<CgroupCommon>();
+        for (File child : files) {
+            if (child.isDirectory()) {
+                children.add(new CgroupCommon(child.getName(), this.hierarchy, this));
+            }
+        }
+        return children;
+    }
+
+    public boolean isRoot() {
+        return isRoot;
+    }
+
+    public Map<SubSystemType, CgroupCore> getCores() {
+        return CgroupCoreFactory.getInstance(this.hierarchy.getSubSystems(), this.dir);
+    }
+
+    public void delete() throws IOException {
+        this.free();
+        if (!this.isRoot) {
+            this.parent.getChildren().remove(this);
+        }
+    }
+
+    private void free() throws IOException {
+        for (CgroupCommon child : getChildren()) {
+            child.free();
+        }
+        if (this.isRoot) {
+            return;
+        }
+        Set<Integer> tasks = this.getTasks();
+        if (tasks != null) {
+            for (Integer task : tasks) {
+                this.parent.addTask(task);
+            }
+        }
+        CgroupUtils.deleteDir(this.dir);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        boolean ret = false;
+        if (o != null && (o instanceof CgroupCommon)) {
+            ret = (((CgroupCommon)o).hierarchy.equals(this.hierarchy)
+                    && ((CgroupCommon)o).name.equals(this.name))
+                    && ((CgroupCommon)o).dir.equals(this.dir);
+
+        }
+        return ret;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + (this.name != null ? this.name.hashCode() : 0);
```

Comment:
you can use lombok plugin to generate these methods. The null checks you are doing in hashCode, should be done in equals method as well.